packages/next/src/server/config.ts

```diff
@@ -666,6 +666,38 @@ function assignDefaults(
     }
   }
 
+  const userProvidedModularizeImports = result.modularizeImports
+  // Unfortunately these packages end up re-exporting 10600 modules, for example: https://unpkg.com/browse/@mui/icons-material@5.11.16/esm/index.js.
+  // Leveraging modularizeImports tremendously reduces compile times for these.
+  result.modularizeImports = {
+    ...(userProvidedModularizeImports || {}),
+    // This is intentionally added after the user-provided modularizeImports config.
+    '@mui/icons-material': {
+      transform: '@mui/icons-material/{{member}}',
+    },
+    '@mui/material': {
+      transform: '@mui/material/{{member}}',
+    },
+    'date-fns': {
+      transform: 'date-fns/{{member}}',
+    },
+    lodash: {
+      transform: 'lodash/{{member}}',
+    },
+    'lodash-es': {
+      transform: 'lodash-es/{{member}}',
+    },
+    'lucide-react': {
+      transform: 'lucide-react/dist/esm/icons/{{ kebabCase member }}',
```

Comment:
`lucide-react` allows importing with an `Icon` suffix like so:
```js
import { HeartIcon, Loader2Icon } from "lucide-react"
```

Which transforms into `heart-icon` and `loader-2-icon` but only `heart.js` and `loader-2.js` exist in `lucide-react/dist/esm/icons/`.

Maybe a new Handlebars helper needs to be created to remove the `Icon`/`-icon` suffix?

Here is the error produced by the current `lucide-react` transform:
```
- error ./src/components/CustomUserButton.tsx:10:0
Module not found: Can't resolve 'lucide-react/dist/esm/icons/loader-2-icon'
   8 |   useUser,
   9 | } from "@clerk/nextjs"
> 10 | import { Loader2Icon } from "lucide-react"
  11 | import { usePathname, useSearchParams } from "next/navigation"
  12 | import { Button } from "~/components/ui/button"
  13 |
  ```

Reply:
Good catch! Let me see how we can handle this.

Reply:
So we decided to comment this one out until we can handle it in our SWC transform. It seems to require a Handlebars helper indeed.